src/components/calcite-label/calcite-label.tsx

```diff
@@ -74,15 +74,25 @@ export class CalciteLabel {
   //--------------------------------------------------------------------------
 
   @Listen("click")
-  onClick(event: MouseEvent): void {
-    const forAttr = this.el.getAttribute("for");
-    this.calciteLabelFocus.emit({
-      labelEl: this.el,
-      interactedEl: event.target as HTMLElement,
-      requestedInput: forAttr
-    });
-    if (forAttr) {
-      document.getElementById(forAttr).click();
+  onClick({ target }: MouseEvent): void {
+    if (
+      (target === this.el) ||
+      (target === this.labelEl) ||
+      (target === this.spanEl)
```

Comment:
This ensures only clicks on the actual calcite-label elements apply, not the bubbled child click events or anything else.